### PR TITLE
Comments on pull files

### LINKED ISSUE #124: Commits and pull files
Before we can start a review, we can implement comment on one line of PRs at first.
I think this includes two sub issues. 

* [ ] ~Commit comment #3748~ see #4898
* [x] Pull files comment #3748

### PR DESCRIPTION
**Pull request by lunny**
First step on comment in pull files, and will resolved #124, but still WIP.

### CODE REVIEW

File: templates/base/head.tmpl [hunk 1]
```diff
@@ -63,7 +63,7 @@
 	<link rel="preload" href="{{AppSubUrl}}/vendor/assets/font-awesome/css/font-awesome.min.css" as="style" onload="this.rel='stylesheet'">
 	<noscript><link rel="stylesheet" href="{{AppSubUrl}}/vendor/assets/font-awesome/css/font-awesome.min.css"></noscript>
 	<link rel="stylesheet" href="{{AppSubUrl}}/vendor/assets/octicons/octicons.min.css">
-
+	<link rel="stylesheet" href="{{AppSubUrl}}/css/main.css">
```
Comment (lafriks) [inline templates/base/head.tmpl:66]: Why new css file? I think these css additions need to be done in less file and built into minimized index.css

**Review by daviian — CHANGES_REQUESTED** [review]

File: models/issue_comment.go [hunk 2]
```diff
@@ -94,7 +96,8 @@ type Comment struct {
 	NewTitle       string
 
 	CommitID        int64
-	Line            int64
+	TreePath        string
+	Line            int64  // + is left; - is right
```
[the remark this answers is not in the document]
Comment (lunny) [reply models/issue_comment.go:100]: Don't want to add an extra column, so I define the line, if it's greater than zero, it should the previous code line before the pull's commits. If it's less than zero, it should the line on this pull's commits.

[the remark this answers is not in the document]
Comment (daviian) [reply models/issue_comment.go:100]: If everyone's happy with this I won't obstruct.

File: public/js/index.js [hunk 3]
```diff
@@ -1831,3 +1833,290 @@ function toggleStopwatch() {
 function cancelStopwatch() {
     $("#cancel_stopwatch_form").submit();
 }
+
+/* global showdown */
+function createCodeComment() {
+    const texts = {
+      btn: "Comment Code",
```
Comment (ethantkoenig) [inline public/js/index.js:1840]: Shouldn't we use locales for these?

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

**Comment by bkcsoft** [comment]
@lunny Got any screenshots? And are there visible when viewing a specific commit? (like here: https://try.gitea.io/gitea/gitea/commit/b7da5a6cb7c725232c1053a1ca705a6ac0dad467 )

File: models/issue_comment.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Morlinest) [reply models/issue_comment.go:100]: IMO, if you use `+/-`, it should be `-` for previous line and `+` for new line.

**Comment by lunny** [comment]
@bkcsoft currently only for pull files. For specific commit should be another PR after this merged.

**Comment by lunny** [comment]
@thehowl yes.

**Pull request closed without merging**